src/Oscoin/Storage/ContentStore.hs

```diff
@@ -0,0 +1,54 @@
+module Oscoin.Storage.ContentStore
+    ( ContentStore
+    , storeContent
+    , lookupContent
+
+    , newIoContentStore
+    , mkStateContentStore
+    , hoistContentStore
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Crypto.Hash as Crypto
+
+import qualified Data.Map as Map
+import           Lens.Micro
+import           Lens.Micro.Mtl
+
+data ContentStore crypto a m = ContentStore
+    { storeContent  :: a -> m ()
+    , lookupContent :: Crypto.Hashed crypto a -> m (Maybe a)
```

Comment:
Name brainstorming: how does `KeyValueStore` sound?

Reply:
I think `KeyValueStore` is confusing because we fixed keys. `ContentStore` comes from “content addressable storage” and the fact that the content determines how we can look it up.